corehq/motech/management/commands/ping_motech_connections.py

```diff
@@ -1,34 +1,204 @@
 import csv
+import os
+from argparse import FileType
+from urllib.parse import urlparse, urlunparse
 
-from django.core.management.base import BaseCommand
-from requests import RequestException, HTTPError
+import requests
+from django.core.management.base import BaseCommand, CommandError
+from oauthlib.oauth2.rfc6749.errors import OAuth2Error
+from requests.exceptions import HTTPError, RequestException, SSLError
 
 from corehq.motech.models import ConnectionSettings
 from corehq.util.log import with_progress_bar
+from corehq.util.urlvalidate.ip_resolver import CannotResolveHost
+
+
+IMPLICIT_HTTPS_PORT = 443
 
 
 class Command(BaseCommand):
-    help = "Make an HTTP HEAD request to all configured MOTECH connections to test the connectivity" \
-           "and SSL validation (if connection setting has not disabled SSL validation)."
+
+    help = "Make an HTTP HEAD request to configured MOTECH connections " \
+           "to test the connectivity of each URL or verify connection domain " \
+           "SSL certificates."
+
+    def __init__(self, *args, **kw):
+        super().__init__(*args, **kw)
+        # disable special styling for STDERR
+        self.stderr.style_func = self.stdout.style_func
+
+    def add_arguments(self, parser):
+        parser.add_argument("-c", "--ca-bundle", metavar="FILE",
+            help="Use a custom CA trust store for SSL verifications.")
+        parser.add_argument("--connect-timeout", metavar="SECONDS", type=float,
+            help="Use custom HTTP connection timeout value.")
+        parser.add_argument("--ssl-only", action="store_true", default=False,
+            help="Skip normal connection checks and perform only SSL "
+                 "verification tests against domains of connections with "
+                 "verify-enabled HTTPS URLs.")
+        parser.add_argument("-o", "--csv-out", metavar="FILE", default="-",
+            type=FileType("w", encoding="utf-8"),
+            help="Write failures to CSV %(metavar)s (default=STDOUT)")
 
     def handle(self, *args, **options):
+        self.verbose = options["verbosity"] > 1
+        timeout = options["connect_timeout"]
+        castore = options["ca_bundle"]
+
+        # sanity-check options
+        if timeout < 0:
+            raise CommandError(f"Invalid timeout value: {timeout}")
```

Comment:
If you want you can use an action to validate the range: https://github.com/dimagi/commcare-hq/blob/354ef5bd94dab77a5bed76c6f46d1502c9e346d0/corehq/util/argparse_types.py#L25